### PR TITLE
Preparation for 1.16.0 release

### PR DESCRIPTION
**Pull request by denisenkom**
Fixes mypy failures

Fix for test_sec_context test

### CODE REVIEW

File: src/pytds/tds_base.py [hunk 1]
```diff
@@ -955,7 +956,7 @@ def __init__(self) -> None:
         self.validate_host = True
         self.enc_login_only = False
         self.enc_flag = 0
-        self.tls_ctx = None
+        self.tls_ctx: None | OpenSSL.SSL.Context = None
```
Comment (mjk4) [inline src/pytds/tds_base.py:959]: allowing pipes for Union types doesn't work in python 3.9: https://peps.python.org/pep-0604/

File: tests/sspi_test.py [hunk 2]
```diff
@@ -44,11 +44,9 @@ def test_sec_context(self):
 
         token_buf = create_string_buffer(10000)
         bufs = [(pytds.sspi.SECBUFFER_TOKEN, token_buf)]
-        server = settings.HOST
-        if "\\" in server:
-            server, _ = server.split("\\")
-        host, _, _ = socket.gethostbyname_ex(server)
-        target_name = "MSSQLSvc/{0}:1433".format(host)
+        # getting current host's name
+        host, _, _ = socket.gethostbyname_ex("")
```
Comment (mjk4) [inline tests/sspi_test.py:48]: IDK about other versions of python, but at least on 3.9 I get a `Resolver Error` when a blank str is given to `socket.gethostbyname_ex`.

**Review by mjk4 — CHANGES_REQUESTED** [review]
Just some py 3.9 compatibility concerns.

File: src/pytds/tds_base.py (hunk 1 above)
Comment (BaoshanGu) [reply src/pytds/tds_base.py:959]: Does this mean 1.16.0 will not work in python 3.9, or this just affects the package building?

File: tests/sspi_test.py (hunk 2 above)
Comment (mjk4) [reply tests/sspi_test.py:48]: Actually, it doesn't seem to work with 3.11, either.

File: src/pytds/tds_base.py (hunk 1 above)
Comment (BaoshanGu) [reply src/pytds/tds_base.py:959]: I see. Thanks

**Review by mjk4 — CHANGES_REQUESTED** [review]
Please fix noted issues.

File: tests/sspi_test.py (hunk 2 above)
Comment (denisenkom) [reply tests/sspi_test.py:48]: Thanks for reviewing, will take a look.
Weirdly tests passed on Python 3.8 https://ci.appveyor.com/project/denisenkom/pytds/builds/50784050

**Comment by Wedge009** [comment]
Not intending to add any pressure, just trying to get a sense of your schedule: when might we expect the next release? I'm interested on the assumption that it will resolve #167/#176.

Failing that, which release was the last one that didn't exhibit this issue?

File: tests/sspi_test.py (hunk 2 above)
Comment (denisenkom) [reply tests/sspi_test.py:48]: This works on Windows on Python 3.8 and 3.12 and this test is only executed on Windows.

File: src/pytds/tds_base.py (hunk 1 above)
Comment (denisenkom) [reply src/pytds/tds_base.py:959]: This module has
```python
from __future__ import annotations
```
Which apparently enables Union type support for Python starting at least from 3.8

For example this code works on Python 3.8:
```python
from __future__ import annotations
class Foo:
    asdf: None | int
```

**Review by mjk4 — APPROVED** [review]
Thanks!

**Pull request merged**